pkg/azurefile/azurefile.go

```diff
@@ -876,6 +887,78 @@ func (d *Driver) ResizeFileShare(ctx context.Context, subsID, resourceGroup, acc
 	})
 }
 
+// CopyFileShare copies a fileshare in the same storage account
+func (d *Driver) copyFileShare(ctx context.Context, req *csi.CreateVolumeRequest, accountKey string, shareOptions *fileclient.ShareOptions, storageEndpointSuffix string) error {
+	if shareOptions.Protocol == storage.EnabledProtocolsNFS {
+		return fmt.Errorf("protocol nfs is not supported for volume cloning")
+	}
+	var volumeID string
+	if req.GetVolumeContentSource() != nil && req.GetVolumeContentSource().GetVolume() != nil {
+		volumeID = req.GetVolumeContentSource().GetVolume().GetVolumeId()
+	}
```

Comment:
what if `volumeID` is empty? and `volumeID` should be `sourceVolumeID`

Reply:
if it is empty, then `GetFileShareInfo` func will return an error and it will return NotFound error at last